### DISCUSSION TITLE
Canvas - overflow hidden behaviour is annoying

### DISCUSSION THREAD

**Discussion by Powerplex** [storybookjs/storybook#25124] [Help]
### Summary

Hey :) I am using Storybook for our design system [Spark](https://sparkui.vercel.app/?path=/docs/experimental-dropdown--docs#default)

We have been doing it for months, but I'm currently implementing a Dropdown component (check the link above). 

The default styling of the canvas is to add overflow: `auto` to everything outside of it, so my dropdown list is cropped. My only solution for now is to add this very ugly white space in every single story just because of this. It also means I must be extra-careful with the content displayed so that it's not higher than what's available.

I know I can use `preview-head` or `preview.css` to override this, but it's very unstable as I have to target three DOM nodes and forces an overflow: visible on them with `!important`, + this could break is SB update its markup in the future.

I can't find documentation for this, is it voluntary, what strategy should I follow for this type of component, I don't want to wrap them all in a portal just for this.

Here is how it behave if I removed my fixed height:

https://github.com/storybookjs/storybook/assets/2033710/8af4ca02-f801-4975-b295-0d7b4c72de27

Here is what I would like, more or less:
![Capture d’écran 2023-12-05 à 20 00 12](https://github.com/storybookjs/storybook/assets/2033710/84c3ecfc-f3db-43f2-8261-2e4158d2f1f4)

**Comment by Marklb** [comment]
This looks like proposed change: #23699

I provided a potential workaround, as a decorator, here: https://github.com/storybookjs/storybook/issues/25083#issuecomment-1837319849
*I don't consider that workaround to be perfect or tested, because I actually prefer my story to be contained in the canvas and know how to adjust itself for the available space. I don't have many components, like that, though, so I can't say I wouldn't like the option for not being contained, which the proposal is for.*

**Comment by RVledder** [comment]
Hi @Powerplex , how did you eventually solve it?

I see that there's a pb-[300px] class that you use to add space below the dropdown.

How have you added that exactly?

---

Ah, a decorator would work :).

**/.storybook/decorators.ts**

```ts
export const dropdownMinHeightWrapper = () => (Story, context) => {
  const story = Story(context);
  return {
    ...story,
    template: `
      <div style="min-h[400px]">
        ${story.template}
      </div>
    `,
  };
}
```

And then in your <component>.stories.ts:

```ts
// ...
import { dropdownMinHeightWrapper } from '../../../.storybook/decorators';

const meta: Meta<DropdownComponent> = {
  component: DropdownComponent,
  title: '/DropdownComponent',
  id: 'DropdownComponent',
  decorators: [dropdownMinHeightWrapper()],
};
export default meta;
type Story = StoryObj<DropdownComponent>;
```
